test/plots/barPlotTests.ts

```diff
@@ -1 +1,783 @@
+            `baseline ${basePositionAttr + "2"} is correct`);
+
+          const bars = barPlot.content().selectAll("rect");
+          assert.strictEqual(bars.size(), data.length, "One bar was created per data point");
+          bars.each(function(datum, index) {
+            const bar = d3.select(this);
+            const baseSize = TestMethods.numAttr(bar, baseSizeAttr);
+            assert.closeTo(baseSize, baseScale.rangeBand(), window.Pixel_CloseTo_Requirement,
+              `bar ${baseSizeAttr} is correct (index ${index})`);
+
+            const valueSize = TestMethods.numAttr(bar, valueSizeAttr);
+            assert.closeTo(valueSize, Math.abs(valueScale.scale(datum.value) - scaledBaselineValue),
+              window.Pixel_CloseTo_Requirement, `bar ${valueSizeAttr} is correct (index ${index})`);
+
+            const basePosition = TestMethods.numAttr(bar, basePositionAttr);
+            assert.closeTo(basePosition, baseScale.scale(datum.base) - 0.5 * baseSize, window.Pixel_CloseTo_Requirement,
+              `bar ${basePositionAttr} is correct (index ${index})`);
+
+            const valuePosition = TestMethods.numAttr(bar, valuePositionAttr);
+            const isShifted = isVertical ? (datum.value > barPlot.baselineValue()) : (datum.value < barPlot.baselineValue());
+            const expectedValuePosition = (isShifted) ? scaledBaselineValue - valueSize : scaledBaselineValue;
+            assert.closeTo(valuePosition, expectedValuePosition, window.Pixel_CloseTo_Requirement,
+              `bar ${valuePositionAttr} is correct (index ${index})`);
+          });
+        }
+
+        it("renders with no data", () => {
+          assert.doesNotThrow(() => barPlot.renderTo(svg), Error);
+          assert.strictEqual(barPlot.width(), TestMethods.numAttr(svg, "width"), "was allocated width");
+          assert.strictEqual(barPlot.height(), TestMethods.numAttr(svg, "height"), "was allocated height");
+        });
+
+        it("draws bars and baseline in correct positions", () => {
+          barPlot.addDataset(dataset);
+          barPlot.renderTo(svg);
+
+          assertCorrectRendering();
+        });
+
+        it("rerenders correctly when the baseline value is changed", () => {
+          barPlot.addDataset(dataset);
+          barPlot.renderTo(svg);
+
+          barPlot.baselineValue(1);
+          assertCorrectRendering();
+        });
+
+        it("can autorange value scale based on visible points on base scale", () => {
+          const firstTwoBaseValues = [data[0].base, data[1].base ];
+          valueScale.padProportion(0);
+          baseScale.domain(firstTwoBaseValues);
+          barPlot.addDataset(dataset);
+          barPlot.autorangeMode(valuePositionAttr);
+          barPlot.renderTo(svg);
+
+          const valueScaleDomain = valueScale.domain();
+          const expectedValueDomainMin = Math.min(data[0].value, data[1].value);
+          const expectedValueDomainMax = Math.max(data[0].value, data[1].value);
```

Comment:
But the point of this test is to check that the autoranging takes place based on the visible data.